### PR TITLE
add BinPastes

### PR DESCRIPTION
**Pull request by querwurzel**
https://github.com/querwurzel/BinPastes

Thanks for taking the time to suggest an addition to awesome-selfhosted! 

To ensure your Pull Request is dealt with swiftly, please check the following (check the boxes `[x]`):
- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted-data/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pulls), including closed ones.
- [x] Any software you are adding is not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] The file you are adding is formatted as described in [addition.md](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/ISSUE_TEMPLATE/addition.md).
- [x] `Demo` links should only be used for interactive demos, i.e. not video demonstrations. 
- [x] Comments and unused optional fields have been removed.
- [x] The file you are adding uses [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) file naming, for example `my-awesome-software.yml`.
- [x] Values for `platform` are the main **server-side** requirements for the software. Don't include frameworks or specific dialects.
- [x] Any software project you are adding to the list is actively maintained.
- [ ] Any software project you are adding was first released more than 4 months ago.
- [x] Any software project you are adding has working installation instructions.
- [x] If you are adding software forked from another active project, please provide/link to a clear list of differences between both.
- [x] You understand that your Pull Request will be merged ~1 week after approval, to allow for further comments if needed.

### CODE REVIEW

**Review by kokomo123 — COMMENTED** [review]
there are no stable release tags present for BinPastes, could you consider making a stable tag for releases so users know that it's stable without building from master branch each time?

File: software/binpastes.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+website_url: https://github.com/querwurzel/BinPastes
+description: Minimal pastebin also supporting client-side encryption, fulltext search, one-time messages. Intended for one to few users looking for a simple pastebin deployment.
+licenses:
+  - Apache-2.0
+platforms:
+  - Java
+tags:
+  - Pastebins
+source_code_url: https://github.com/querwurzel/BinPastes
+stargazers_count: 2
```
Comment (kokomo123) [inline software/binpastes.yml:11]: These will be filled in automatically, so feel free to remove them. 
```suggestion
- stargazers_count: 2
- updated_at: '2023-11-13'
- archived: false
```

**Comment by querwurzel** [comment]
> there are no stable release tags present for BinPastes, could you consider making a stable tag for releases so users know that it's stable without building from master branch each time?

Fair enough. Done. Will add an automation for releases soon.

**Comment by querwurzel** [comment]
@nodiscc the first commit is from June this year although I'm working on it since december last year. So the 4 months period is long over.

**Comment by nodiscc** [comment]
Hi,

>  Any software project you are adding was first released more than 4 months ago.

The [first release](https://github.com/querwurzel/BinPastes/releases) must be 4 months old, feel free to resubmit after 14 March 2024. Thanks

**Pull request closed without merging**